Fix `COLUMNS(* REPLACE)` generating duplicate columns with joins

tobwen opened the pull request:
When using `COLUMNS(* REPLACE column AS column)` with joins, the column that appears in both tables (the `USING` column) was being replaced multiple times - once for each table. This resulted in duplicate columns in the output instead of a single merged column.

### Example
``` sql
SELECT COLUMNS(* REPLACE a AS a)
FROM range(1) t1(a) NATURAL JOIN range(1) t2(a)
```

### Summary
The fix tracks which columns have already been replaced in the ExclusionListInfo structure to prevent duplicate replacement expressions from being added for USING columns.

### Reference issue
fixes #18604

tobwen committed:
Fix `COLUMNS(* REPLACE)` generating duplicate columns with joins

When using `COLUMNS(* REPLACE column AS column)` with joins, the column that appears in both tables (the `USING` column) was being replaced multiple times - once for each table. This resulted in duplicate columns in the output instead of a single merged column.

# example

``` sql
SELECT COLUMNS(* REPLACE a AS a)
FROM range(1) t1(a) NATURAL JOIN range(1) t2(a)
```

The fix tracks which columns have already been replaced in the ExclusionListInfo structure to prevent duplicate replacement expressions from being added for USING columns.

Mytherin commented:
Thanks for the PR!

I think it would be better to handle this in the same way that we handle this for renames. Perhaps we can move the replace expression code to `HandleRename` instead of handling this in the exclusion list?

tobwen committed:
Refactoring to `HandleRename`

Moves the replace expression code to `HandleRename` instead of handling this in the exclusion list. Also added some tests.

tobwen commented:
> Perhaps we can move the replace expression code to `HandleRename` instead of handling this in the exclusion list?

Complicated... but done.
